crates/database/src/quotes.rs

```diff
@@ -136,12 +140,88 @@ pub async fn remove_expired_quotes(
     const QUERY: &str = r#"
 DELETE FROM quotes
 WHERE expiration_timestamp < $1
+RETURNING id
     "#;
-    sqlx::query(QUERY)
+    let deleted_ids = sqlx::query(QUERY)
         .bind(max_expiry)
+        .fetch_all(&mut *ex)
+        .await?
+        .iter()
+        .map(|row| row.get(0))
+        .collect::<Vec<_>>();
+
+    delete_quote_interactions(ex, &deleted_ids).await
+}
+
+/// One row in the `quote_interactions` table.
+#[derive(Clone, Debug, PartialEq, sqlx::FromRow)]
+pub struct QuoteInteraction {
+    pub quote_id: QuoteId,
+    pub index: i32,
+    pub target: Address,
+    pub value: BigDecimal,
+    pub call_data: Vec<u8>,
+}
+
+/// Stores interactions provided by the solver for quote.
+pub async fn insert_quote_interaction(
+    ex: &mut PgConnection,
+    quote_interaction: &QuoteInteraction,
+) -> Result<(), sqlx::Error> {
+    const QUERY: &str = r#"
+INSERT INTO quote_interactions (
+    quote_id,
+    index,
+    target,
+    value,
+    call_data
+)
+VALUES ($1, $2, $3, $4, $5)
+    "#;
+    sqlx::query(QUERY)
+        .bind(quote_interaction.quote_id)
+        .bind(quote_interaction.index)
+        .bind(quote_interaction.target)
+        .bind(&quote_interaction.value)
+        .bind(&quote_interaction.call_data)
         .execute(ex)
-        .await
-        .map(|_| ())
+        .await?;
+    Ok(())
+}
+
+pub async fn insert_quote_interactions(
+    ex: &mut PgConnection,
```

Comment:
```suggestion
    ex: &mut PgTransaction,
```
That way, we make sure there is an open tx.